### PR TITLE
chore: cleanup dependencies

### PR DESCRIPTION
**Pull request by frimoldi**
- Remove Apollo
- Upgrade wagmi from `v0.2` to `v0.5`

### CODE REVIEW

File: src/components/ConnectButton/ConnectButton.tsx [hunk 1]
```diff
@@ -14,17 +14,27 @@ import WalletProviderModal from "../WalletProviderModal/WalletProviderModal"
...
+  const [isCorrectNetwork, setIsCorrectNetwork] = useState(false)
 
-  const [{ data: connectionData }] = useConnect()
-  const [{ data: networkData }, switchNetwork] = useNetwork()
-  const [{ data: accountData }] = useAccount()
+  const { chain } = useNetwork()
+  const { switchNetwork } = useSwitchNetwork()
+  const { address: connectedAddress, isConnected } = useAccount()
+
+  console.log(isConnected)
```
Comment (rcstanciu) [inline src/components/ConnectButton/ConnectButton.tsx:23]: I think this can be removed

File: src/pages/Claim/StartNewClaimSection.tsx [hunk 2]
```diff
@@ -6,22 +6,19 @@ import { Text } from "../../components/Text"
...
+  const { address: connectedAddress } = useAccount()
   const [isCreating, setIsCreating] = useState(false)
   const [canStartNewClaim, setCanStartNewClaim] = useState(false)
 
   useEffect(() => {
-    setCanStartNewClaim(
-      !!connectedAccount?.address && ethers.utils.getAddress(connectedAccount.address) === protocol.agent
-    )
-  }, [connectedAccount?.address, protocol.agent])
+    setCanStartNewClaim(!!connectedAddress && connectedAddress === protocol.agent)
```
Comment (rcstanciu) [inline src/pages/Claim/StartNewClaimSection.tsx:20]: Here you are using double negation to check if `connectedAddress` has a *truthy* value, but in the other files, you did not check this. It's redundant, the comparison will still work without this check, but wanted to know if there should be a difference between this file and the others.

File: src/utils/WagmiProvider.tsx [hunk 3]
```diff
@@ -19,37 +17,38 @@ const chains = defaultChains
...
-        rpc: {
-          [networkId]: alchemyApiUrlHttp,
-        },
-      },
+const { provider, webSocketProvider } = configureChains(chains, [
+  alchemyProvider({ alchemyId: alchemyApiKey }),
+  publicProvider(),
+  jsonRpcProvider({
+    rpc: () => ({
+      http: "http://127.0.0.1:8545",
```
Comment (rcstanciu) [inline src/utils/WagmiProvider.tsx:29]: Does this fix now the annoying errors popping up when running against a local node?

**Review by rcstanciu — APPROVED** [review]
Looking good to me

File: src/utils/WagmiProvider.tsx (hunk 3 above)
Comment (frimoldi) [reply src/utils/WagmiProvider.tsx:29]: Yeah, I'm not getting those messages anymore 🙏🏽

File: src/pages/Claim/StartNewClaimSection.tsx (hunk 2 above)
Comment (frimoldi) [reply src/pages/Claim/StartNewClaimSection.tsx:20]: Nope, you're right. I can remove the first check.

**Pull request merged**